data_depot_downloader/utils/metadata_service.py

```
@@ -7,11 +7,46 @@ def __init__(self, *, base_url, username, password):
         self._username = username
         self._password = password
 
-    def get_live_dataset_info(self):
+    def get_live_dataset_oids(self):
+        # Make an api request to the metadata service
+        # for all live dataset ids
         response = self._get('/datasets/live')
         response.raise_for_status()
         return response.json()['live_datasets']
 
+    def get_dataset_download_creds(self, oid):
+        response = self._get(
+            f'/datasets/{oid}/connection-info',
+        )
+        response.raise_for_status()
+        return response.json()['dataset_creds']
+
+    def get_source_file_creds(self, oid, md5):
+        # Make an api request to the metadata service
+        response = self._get(f'/datasets/{oid}/source_file/upload-credentials'
+                             f'?file_md5={md5}')
+        response.raise_for_status()
+        if response.status_code == 204:
+            return {}
```

Instead of returning an `{}` maybe we can raise an error, and then catch the error in the caller so it is a bit more informative about what happened here